app/src/main/java/io/github/gsantner/memetastic/activity/MemeCreateActivity.java

```diff
@@ -449,24 +459,22 @@ private void initMoarControlsContainer() {
         //
         View.OnClickListener colorListener = new View.OnClickListener() {
             public void onClick(View v) {
-                LineColorPicker picker = (LineColorPicker) v;
-                if (picker == colorPickerShade) {
-                    _memeEditorElements.getCaptionTop().setBorderColor(picker.getColor());
-                    _memeEditorElements.getCaptionBottom().setBorderColor(picker.getColor());
-                } else if (picker == colorPickerText) {
-                    _memeEditorElements.getCaptionTop().setTextColor(picker.getColor());
-                    _memeEditorElements.getCaptionBottom().setTextColor(picker.getColor());
+                LinearLayout picker = (LinearLayout) v;
+                if (picker == colorPickerTextBorder) {
+                    ColorPickerDialog.newBuilder().setDialogId(1).setColor(_memeEditorElements.getCaptionTop().getBorderColor()).show(MemeCreateActivity.this);
+                } else if (picker == colorPickerTextBackground) {
+                    ColorPickerDialog.newBuilder().setDialogId(2).setColor(_memeEditorElements.getCaptionTop().getTextColor()).show(MemeCreateActivity.this);
                 } else if (picker == colorPickerPadding) {
-                    _memeEditorElements.getImageMain().setPaddingColor(picker.getColor());
-                    _memeEditorElements.getImageMain().setPaddingColor(picker.getColor());
+                    ColorPickerDialog.newBuilder().setDialogId(3).setColor(_memeEditorElements.getImageMain().getPaddingColor()).show(MemeCreateActivity.this);
```

Comment:
I would use the ressource id here as parameter, than you can switch on R.id.something, and dont have to introduce new IDs :)

Reply:
okay